algorithms/kernel/dtrees/forest/oneapi/df_feature_type_helper_oneapi.i

```diff
@@ -0,0 +1,567 @@
+/* file: df_feature_type_helper_oneapi.i */
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  GPU-dependent initialization of service data structure
+//--
+*/
+#include "algorithms/kernel/dtrees/dtrees_feature_type_helper.h"
+
+#include "service/kernel/service_data_utils.h"
+#include "externals/service_ittnotify.h"
+
+DAAL_ITTNOTIFY_DOMAIN(df.common.oneapi);
+
+using namespace daal::oneapi::internal;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace decision_forest
+{
+namespace internal
+{
+template <typename FPType>
+struct GetIntegerTypeForFPType;
+
+template <>
+struct GetIntegerTypeForFPType<float>
+{
+    using Type = uint32_t;
+};
+
+template <>
+struct GetIntegerTypeForFPType<double>
+{
+    using Type = uint64_t;
+};
+
+template <typename IntType>
+services::String getOpenCLKeyType(const services::String & typeName);
+
+template <>
+inline services::String getOpenCLKeyType<uint32_t>(const services::String & typeName)
+{
+    return services::String(" -D ") + typeName + services::String("=uint ");
+}
+
+template <>
+inline services::String getOpenCLKeyType<uint64_t>(const services::String & typeName)
+{
+    return services::String(" -D ") + typeName + services::String("=ulong ");
+}
+
+template <typename algorithmFPType>
+static void __buildProgram(ClKernelFactoryIface & factory)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.buildProgram);
+
+    {
+        auto fptype_name    = getKeyFPType<algorithmFPType>();
+        auto radixtype_name = getOpenCLKeyType<typename GetIntegerTypeForFPType<algorithmFPType>::Type>("radixIntType");
+        auto build_options  = fptype_name + radixtype_name;
+        build_options.add("-cl-std=CL1.2");
+
+        services::String cachekey("__daal_algorithms_df_common_");
+        cachekey.add(fptype_name);
```

Comment:
It's possible to replace 2 line with  cachekey.add(build_options);

Reply:
fixed